src/connection/eagerlyConnect.ts

```diff
@@ -26,11 +41,15 @@ const selectedWallet = toConnectionType(localStorage.getItem(selectedWalletKey)
 if (selectedWallet) {
   const selectedConnection = getConnection(selectedWallet)
   if (selectedConnection) {
-    connect(selectedConnection.connector, selectedWallet).then((connected) => {
-      if (!connected) {
-        // only clear the persisted wallet type if it failed to connect.
-        localStorage.removeItem(selectedWalletKey)
-      }
-    })
+    connectionReady = connect(selectedConnection.connector, selectedWallet)
```

Comment:
do you need to set `connectionReady = Promise<void>`? Could you instead `let connectionReady = false` and then just update it to `true` on a successful connect? That would simplify the `if` statement in `useConnectionReady`